### PR TITLE
Md/phys tests

### LINKED ISSUE #507: Extension of tests across PROTEUS ecosystem
We need to substantially expand the test suite across the entire ecosystem. It is not yet quite clear to me what the best way forward is, please suggest ideas, I will then edit and update this posts.

So far, I believe we need (with questions to be decided):
- Unit tests in PROTEUS for all major functions in PROTEUS (which are critical, which are not though?)
- Integration tests (which parts should be tested?)
- Unit tests for all major modules
- The documentation then needs to be extended to describe the tests that are going on (so that it is understandable what is tested where and how to fix it if it breaks)

### PR DESCRIPTION
**Pull request by MarijnJ0**
## Description
This PR introduces new and improved tests, as well as several bug fixes. With these changes the solid1d* models now closely agree on tidal dissipation and Love numbers in the regime where they are stable. The test output is now visible in the docs and README through new badges. In addition to this, I have added the themes from AGNI into Obliqua, and made several changes to get the dark theme to work (it is still not perfect, but looks ok). 

Specifically, this PR adds the following:
- Added safety check for load number and scaling factor.
- Fixed surface boundary condition used by solid1d* models.
- Fixed sign convention between core solution / A matrix / surface boundary used by solid1d* models.
- Fixed issue where the core solution for relaxation solvers would be assumed non-singular before computing the left over nullspace. The computation is now valid for all core solution vectors (removes rigid core rather than a free-slipping one.)
- Added test that verifies compatibility of core and surface boundary.
- Added test that verifies compatibility of core and A matrix.
- Added new equilibrium tide model (Saito 1974) for inhomogeneous sphere: solid1d_equil_relax.
- Added non-dimensional scaling to all solid1d* solvers, it is needed to determine the loading Love number as the matrix objects at the surface become rank deficient.
- Fixed bug where first interior layer was always skipped.
- Fixed bug where shooting method based solid1d models always skipped first layer passed to them.
- Added tests that verify the homogeneous sphere limit, all solid1d models now closely agree with the analytical solution in solid0d. Additionally this checks that in the equilibrium limit the output converges to solid1d_equil_relax.
- Added tests that verifies the solid core solution vector in the homogeneous limit. 
- Added option to use lowest mantle properties (shear/bulk) instead of those from the config. Usefull, for testing, or when Obliqua resolves the core and the 'core solution vector' merely acts as a guard against the singularity at r=0.
- Obliqua now explicitely tracks all dissipation channels (shear, bulk, Darcy, friction, drag) as 3d objects. 
- fluid1d now has an updated radial distribution model that smoothly attaches to 3d dissipation objects (shear, bulk, Darcy), and extends the heating profile with isotropic friction and drag.
- Added tests.jl badges.
- Added light and dark themes for the docs, similar to those used by PROTEUS.

## Still to add
I have already some updated plotting scripts, and will add them in the upcomming days. First I want to work on the orbit/PROTEUS part. 

I want to add some additional tests for the mush models to further verify the code.

The inertial core solution vector needs to be further looked into, I have found a good source. 

Let me know if I missed something! :)

## Validation of changes
afbeelding
afbeelding
afbeelding
afbeelding
afbeelding
afbeelding
afbeelding
afbeelding
afbeelding

See the dissipation channel example plot in the overleaf document.

## Checklist

- [x] I have followed the [contributing guidelines](https://fwl-proteus.readthedocs.io/en/latest/CONTRIBUTING.html#how-do-i-contribute)
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings or errors
- [x] I have checked that the tests still pass on my computer
- [x] I have updated the docs, as appropriate
- [x] I have added tests for these changes, as appropriate
- [x] I have checked that all dependencies have been updated, as required

## Relevant people
@nichollsh @timlichtenberg

### CODE REVIEW

File: src/solid1d.jl [hunk 1]
```diff
@@ -71,14 +71,20 @@ module solid1d
         g = zeros(prec, size(r))
         M = zeros(prec, size(r))
 
-        for i in 1:size(r)[2]
-            M[2:end,i] = 4.0/3.0 * π .* diff(r[:,i].^3) .* ρ[i]
+        # Base mass enclosed at the inner boundary of the first layer (Core Mass)
+        M[1, 1] = m_core
+
+        # Shell mass incremental calculations
+        for i in 1:size(r, 2)
```
Comment (nichollsh) [inline src/solid1d.jl]: Worthwhile adding a check or numerical offset here to ensure that `r=0` never occurs. This might happen for coreless planets, or if we eventually resolve the planet's core in 1D.

File: src/solid1d.jl [hunk 2]
```diff
@@ -372,9 +403,9 @@ module solid1d
 
         M[1, :] .= y1_4[3,:,end,end]  # Row 1 - Radial Stress 
         M[2, :] .= y1_4[4,:,end,end]  # Row 2 - Tangential Stress
-        M[3, :] .= y1_4[6,:,end,end] .+ (n+1)/r[end:end] .* y1_4[5,:,end,end]  # Row 3 - Potential Stress
```
Comment (nichollsh) [inline src/solid1d.jl]: Why remove these extra terms?

File: src/solid1d.jl [hunk 3]
```diff
@@ -390,44 +421,40 @@ module solid1d
...
-        b[1] = -g[end,end] * zeta * G / r[end,end] - P
-        b[2] = tau
-        b[3] = ((2 * n + 1) / r[end,end]) * U + 4 * pi * G * zeta
-
-        C = M \ b
-
+        # Compute the boundary conditions at the surface
+        _, b = get_surface_bc!(r[end,end]/scale[1], g[end,end]/scale[6], n, U, U_prime, tau, P; G0=scale[5], Y=[1,2,3,4,5,6])
+        
+        C = M \ b[collect((3, 4, 6))]
```
Comment (nichollsh) [inline src/solid1d.jl]: What is `C` here?

File: src/solid1d_mush_relax.jl [hunk 4]
```diff
@@ -928,148 +928,33 @@ module solid1d_mush_relax
...
-    - `ρ::prec`                          : Average core density.
-    - `g::prec`                          : Gravity at the core-mantle boundary.
-    - `μ::prec`                          : Average core shear modulus.
-    - `K::prec`                          : Average core bulk modulus.
-    - `type::String`                     : Type of core boundary condition to apply. Options are "liquid" for a fluid core, "solid" for a solid core, and "inertial" for a core with inertial response.
-    - `n::Int`                           : Tidal degree.
+    - `ω::Float64`                        : Forcing frequency.
+    - `r::prec`                           : Radial position of the core-mantle boundary.
+    - `ρ::prec`                           : Average core density.
+    - `g::prec`                           : Average core gravity.
```
Comment (nichollsh) [inline src/solid1d_mush_relax.jl]: Are you sure that this is really the average core gravity? The way it is used does imply that it's the gravity at the CMB.

File: src/solid1d_equil_relax.jl [hunk 5]
```diff
@@ -0,0 +1,519 @@
...
+    Tuple of resampled profiles on the new grid:
+    - `r_new_b::Vector{prec}`             : New radius profile at layer boundaries.
+    - `ρ_new::Vector{prec}`               : New density profile at layer centers.
+    - `η_new::Vector{prec}`               : New viscosity profile at layer centers.
+    - `μ_new::Vector{precc}`              : New shear modulus profile at layer centers.
+    - `κ_new::Vector{precc}`              : New bulk modulus profile at layer centers.
+    - `g_new::Vector{prec}`               : New gravity profile at layer centers.
+    - `M_tot::Float64`                    : Total mass enclosed within the outermost layer boundary.
+    """ 
+    function resample_profiles(radius::Vector{prec}, rho::Vector{prec}, visc::Vector{prec}, shear::Vector{precc}, bulk::Vector{precc}, m_core::prec, dr_min::Int64, dr_max::Int64)
```
Comment (nichollsh) [inline src/solid1d_equil_relax.jl]: This function `resample_profiles` seems to be duplicated in multiple files for each of the solid1d models. Doing this makes the code hard to understand, maintain, and test. Please remove duplicated functions by putting a single definition into a shared file.

File: src/solid1d_equil_relax.jl [hunk 6]
```diff
@@ -0,0 +1,519 @@
...
+    # Arguments
+    - `r::Array{prec,1}`               : 1D array of layer boundaries. 
+    - `ρ::Array{prec,1}`               : 1D array of layer densities. The length of `ρ` must be equal to the number of columns in `r`.
+    - `m_core::prec`                   : Mass of the core.
+
+    # Returns
+    - `g::Array{prec,1}`               : 1D array of gravity values at the layer boundaries. The dimensions of `g` are the same as `r`.
+    - `M_enc::prec`                    : Total mass enclosed within the outermost layer boundary.
+    """
+    function get_g(r::Vector{prec}, ρ::Vector{prec}, m_core::prec)
```
Comment (nichollsh) [inline src/solid1d_equil_relax.jl]: There are multiple `get_g` functions in the code - one per tidal model, I believe. Is this really necessary, or could you have a single shared function?

File: src/solid1d_relax.jl [hunk 7]
```diff
@@ -411,107 +391,32 @@ module solid1d_relax
...
-    - `ρ::prec`                          : Average core density.
-    - `g::prec`                          : Gravity at the core-mantle boundary.
-    - `μ::prec`                          : Average core shear modulus.
-    - `K::prec`                          : Average core bulk modulus.
-    - `type::String`                     : Type of core boundary condition to apply. Options are "liquid" for a fluid core, "solid" for a solid core, and "inertial" for a core with inertial response.
-    - `n::Int`                           : Tidal degree.
+    - `ω::prec`                           : Forcing frequency.
+    - `r::prec`                           : Radial position of the core-mantle boundary.
+    - `ρ::prec`                           : Average core density.
+    - `g::prec`                           : Average core gravity.
```
Comment (nichollsh) [inline src/solid1d_relax.jl]: Not CMB gravity?

**Review by timlichtenberg — CHANGES_REQUESTED** [review]
Great progress on the tests and the theming, this is a lot of work. A few questions on the solver changes before I can approve.

`get_A!` in `common.jl` still takes `ω` as a required argument, but the two `-ω^2*ρ` inertial terms are commented out, not removed. Was that on purpose? The new `Ic` vs `A` consistency test can't catch this either way, since `get_Ic`'s `"solid"` branch has no `ω` term.

Zero forcing frequency now reroutes to `solid1d-equil-relax` whenever `module_solid` contains `"solid1d"`, which also matches `"solid1d-mush"`. Doesn't that send mush segments to the plain solid-body model at σ=0, dropping the mush arguments?

The old surface condition combined `y6 + (n+1)/r*y5`. The new `get_surface_bc!` only sets `y6`. Is that a basis change that makes the `y5` term redundant, or did it get dropped?

`get_core_bc!` moved from `pinv` to `nullspace`, which is the headline boundary-condition fix, but the old unit test for it (checking `B`'s shape and entries) was deleted with nothing in its place. Worth a direct test on the new implementation?

`get_g` divides by `r^2`, including at `r=0`. The mush test grid moved from `r=0.0` to `r=0.1` in this PR. Is that working around the same division rather than fixing it?

Small one: the docs build is failing in CI. `DocumenterTools` is in `docs/Project.toml` now, but `documentation.yml` never installs from that project file, so `docs/make.jl`'s `using DocumenterTools` errors out.

**Review by nichollsh — CHANGES_REQUESTED** [review]
## Summary 

Awesome work, @MarijnJ0 - thanks for this.

The bugfix changes to the code seem reasonable to me, although some of the sections are still light on in-line comments, which makes it hard to understand all of the linear algebra that's implemented. I assume the (many) sign-changes you've included in this PR are some of the key bug fixes in the core model? I appreciate you adding more docstrings throughout the codebase, though.

The tests are great. They seem to include a range of unit, smoke, and integration tests. You've even included comparisons against 'classic' tidal models in the appropriate endmember regimes (e.g. fully liquid tides). 

I ran the tests on my laptop (Fedora 44 on AMD, Julia 1.12.6) and they all pass fine. I am able to use the coverage script which 82% coverage. The docs also build without errors and render fine.

## Changes required

1. The test count badges are created, although they are all labelled as unit tests. The integrate test badge shows zero. Is this correct?

2. The documentation shows the same badge three times. Are these meant to be the three test-type badges? The documentation workflow is also failing because of a missing `DocumenterTools` package.

3. The link to the AGNI documentation is broken in the Obliqua docs 'ecosystem' page. 

4. The reference tables explaining config parameters should go into the Reference section of the docs, not within Tutorial 2.

5. The tutorials do not really feel like a tutorial still. They feel like a series of commands that the user runs without knowing what they do or achieve. Please at least include some figures to show what is outputted, and explain these in the tutorial pages.

6. The "How-To guides" page is empty and the new `solid1d_equil_relax` model is missing a documentation entry.

7. Please add `coverage.*`, `src/*.cov`, and `docs/build/` to the `.gitignore`, to avoid uploading them to GitHub.

8. See my additional in-line comments below.

File: src/solid1d.jl (hunk 3 above)
Comment (MarijnJ0) [reply src/solid1d.jl]: In the shooting method we propagate three independent elementary solutions to the surface, where we have a known state of the true modal solution vector. At this step we compute the weights of integration (one per elemenatry solution). C is the vector containing these weights, i.e. C = [a1, a2, a3]. Hence we multiply y1_4 by C to obtain the modal solution y. In the relaxation method these weights are eliminated right at the core, hence they never show up. I added an in-line comment.

File: src/solid1d.jl (hunk 1 above)
Comment (MarijnJ0) [reply src/solid1d.jl]: Added an offset at the start of run_tides and a check in all run_solid1d* that catches r=0.

File: src/solid1d.jl (hunk 2 above)
Comment (MarijnJ0) [reply src/solid1d.jl]: This is because of a mistake from before. I had originally confused two conventions. We should not include `y5` in the surface boundary given our definition of `y6` (see overleaf). 

This was introduced in the more recent works to simplify the surface boundary conditions.

Additionally, I changed the normalization of the loading potential (`G/G0 / R * U_prime`). The latter does not apply to the equilibrium model, but since it relies on y7 = y6 + 4πG/g y2 it is always zero, no matter the normalization.

With these changes, the boundary conditions now match those used in Sabadini 2016.

**Comment by MarijnJ0** [comment]
Thanks for the feedback! To answer your questions:

> the two -ω^2*ρ inertial terms are commented out, not removed.

This was a typo. Since the solid0d limit is only a valid approximation of solid1d when the inertial terms are removed from the motion matrix, I had commented them out during testing. I added the inertial_terms option to enable/disable inertial terms in motion matrix. This is useful for the shooting method, as it tends to diverge more quickly with these terms enabled. The relaxation based solver handles them no problem.

> Doesn't that send mush segments to the plain solid-body model at σ=0,

No, the solid1d_mush models are simply solid tide models that also capture the Darcy dissipation channel, which is only active in partial melts (mush). As such, using the solid1d_mush models requires the user to merge the solid and mush regimes down to some user defined viscosity. When using the relaxation based solver we can merge the entire solid and mush regimes. Regardless of the choice, the distinct tidal dissipation channels all turn off in the equilibrium limit, hence Im(k_n) = 0 by definition. The deformation is purely elastic, and the Re(k_n) depends merely on the mass distribution (i.e. the density profile). This means that solid1d_equil_relax is the correct simplified version of all other solid1d models. I suppose that if we drop the assumption of homogenity in the fluid regime, then the equations would also reduce to solid1d_equil_relax, Potentially we should apply solid1d_equil_relax in the entire mantle, but we would be mixing assumptions at this point, hence I only added it in place of the solid1d (inhomogeneous) models.

> The old surface condition combined y6 + (n+1)/r*y5.

This is because of a mistake from before. I had originally confused two conventions. We should not include y5 in the surface boundary given our definition of y6 (see overleaf). This was introduced in the more recent works to simplify the surface boundary conditions. Additionally, I changed the normalization of the loading potential (G/G0 / R * U_prime). The latter does not apply to the equilibrium model, but since it relies on y7 = y6 + 4πG/g y2 it is always zero, no matter the normalization. With these changes, the boundary conditions now match those used in Sabadini 2016.

> Worth a direct test on the new implementation?

I added a test that checks if B is the nullspace through B * Ic = 0.

> Is that working around the same division rather than fixing it?

The solid1d models all have a singularity at r=0, not just due to the gravity calculation, but also in the motion matrices. Hence the analytically derived core solution vectors. They literally allow you to set r as small as you like > 0 (you dont have to use them to 'model' the entire metal core, we just do so because we dont have the inner core radial profiles in obliqua yet). So that, say the first layer is set to start at r = 0, then we must at least set r --> r = 0 + epsilon for some tiny epsilon. Since the first layer skip bugs got fixed I had to change the r=0 to r=0.1. In the main run_tides function I now added a check that sets r[1] --> 1e-6 * r[2] if r[1] = 0. As well as, several checks in the run_solid wrapper functions to catch this.

> DocumenterTools

I added it to the build script, it should work now.

> integrate test badge shows zero. Is this correct?

Yes, I did not add these tests yet. But copied over the badges etc, so that it will automatically grab these once they are added.

Note, I changed the runtests.yml file temporarly, but will add back the if statement:

`if: ${{(github.event_name == 'push') && (github.ref == 'refs/heads/main')}}`

> same badge three times. Are these meant to be the three test-type badges?

Yes they should be different: unit, integ, unit+integ.

> The link to the AGNI documentation is broken in the Obliqua docs 'ecosystem' page.

I have changed it to the one used on the PROTEUS site: https://www.h-nicholls.space/AGNI/ (please lmk if this is incorrect.). Additionally, I changed the rest to the proteus-framework.org/ urls.

> config parameters should go into the Reference

Moved the description to the reference, kept the read command in the tutorial.

> Please at least include some figures to show what is outputted, and explain these in the tutorial pages.

> The "How-To guides" page is empty

This will take some time, I will try improving the docs over the next few days.

> the new solid1d_equil_relax model is missing a documentation entry.

Added a new page for this.

> Please add coverage.*, src/*.cov, and docs/build/ to the .gitignore

Done.

I will wait for the tests to finish, and see if the badges get updated...

**Comment by MarijnJ0** [comment]
Docs preview at https://proteus-framework.org/Obliqua/previews/PR31/

The badges where created and look correct. I'll change the runtests.yml back now.

**Comment by nichollsh** [comment]
@MarijnJ0 please request a new review when you are ready

File: docs/src/index.md [hunk 8]
```diff
@@ -1,42 +1,165 @@
...
+### Get started
+
+```@raw html
+    <table width="100%" style="border-collapse: separate; border-spacing: 10px;">
+    <tr>
+        <td width="33%" valign="top" style="border: 1px solid #30363d; border-radius: 8px; padding: 16px; text-align: left;">
+        <h5 style="margin-top: 0; text-align: left;">📥 Install Obliqua</h5>
+        <p style="text-align: left;">Set up the package and its dependencies within the Julia environment.</p>
+        <div style="margin-top: 16px; text-align: left;">
+            <a href="https://proteus-framework.org/Obliqua/dev/how-to-guides/install.html"><b>Installation guide →</b></a>
```
Comment (nichollsh) [inline docs/src/index.md]: This should locally reference the install guide, rather than going to proteus-framework.org.

File: docs/src/index.md [hunk 9]
```diff
@@ -1,42 +1,165 @@
...
+        </div>
+        </td>
+        <td width="33%" valign="top" style="border: 1px solid #30363d; border-radius: 8px; padding: 16px; text-align: left;">
+        <h5 style="margin-top: 0; text-align: left;">🎓 New to Obliqua?</h5>
+        <p style="text-align: left;">Start with the quick-start guide to calculate basic tidal Love numbers.</p>
+        <div style="margin-top: 16px; text-align: left;">
+            <a href="https://proteus-framework.org/Obliqua/dev/tutorials/0d_test.html"><b>Quick start tutorial →</b></a>
```
Comment (nichollsh) [inline docs/src/index.md]: This should locally reference the tutorial, rather than going to proteus-framework.org.

File: docs/src/index.md [hunk 10]
```diff
@@ -1,42 +1,165 @@
...
+        </div>
+        </td>
+        <td width="33%" valign="top" style="border: 1px solid #30363d; border-radius: 8px; padding: 16px; text-align: left;">
+        <h5 style="margin-top: 0; text-align: left;">🌍 Ready for science?</h5>
+        <p style="text-align: left;">Run full planet-mantle dissipation models integrated with PROTEUS.</p>
+        <div style="margin-top: 16px; text-align: left;">
+            <a href="https://proteus-framework.org/Obliqua/dev/tutorials/1d_test.html"><b>Science examples →</b></a>
```
Comment (nichollsh) [inline docs/src/index.md]: This should locally reference the example, rather than going to proteus-framework.org.

File: docs/make.jl [hunk 11]
```diff
@@ -1,58 +1,93 @@
...
+                joinpath(ASSETS_DIR, "themes/documenter-dark.css"))
+
 format = Documenter.HTML(
-    edit_link = "main",  # branch name for "Edit on GitHub"
+    edit_link = "main",
+    collapselevel = 1,
+    size_threshold = 300 * 1024,
+    size_threshold_warn = 200 * 1024,
     prettyurls = get(ENV, "CI", nothing) == "true",
+    assets = [
```
Comment (nichollsh) [inline docs/make.jl]: Please set the favicon to the PROTEUS glyph. See here: https://documenter.juliadocs.org/stable/lib/internals/writers/#Documenter.HTMLWriter.HTML

File: docs/src/index.md [hunk 12]
```diff
@@ -1,42 +1,165 @@
...
-* `src/`                - Source files
-* `docs/`               - Documentation files
-* `out/`                - Output files
-* `res/`                - Resources
-* `test/`               - Tests for the code
-* `examples/`           - Tools and notebooks showing how to use the code
+The repository structure is laid out below. The main components of the repository are the source code, documentation, tests, and examples. The source code is located in the `src/` directory, while the documentation is located in the `docs/` directory. The tests are located in the `test/` directory, and the examples are located in the `examples/` directory.
+
+```bash
+Obliqua/
```
Comment (nichollsh) [inline docs/src/index.md]: Not sure how useful this file tree is, to be honest... maybe remove it?

File: docs/src/index.md [hunk 13]
```diff
@@ -1,42 +1,165 @@
...
 
+---
+
 ### Integrations
 
 This project is being developed in close collaboration with the [PROTEUS](https://github.com/FormingWorlds/PROTEUS) similation framework. Obliqua can be used seamlessly within PROTEUS to model tidal dissipation and Love number evolution in planetary interiors.
 
-For detailed instructions on installing and using Obliqua in combination with PROTEUS, please refer to the [installation guide](https://proteus-framework.org/Obliqua/dev/install/) in the Obliqua documentation.
\ No newline at end of file
+For detailed instructions on installing and using Obliqua in combination with PROTEUS, refer to the [installation guide](https://proteus-framework.org/Obliqua/dev/how-to-guides/install.html) in the Obliqua documentation.
```
Comment (nichollsh) [inline docs/src/index.md]: Broken URL. Again, please use local links in the Obliqua docs.

File: docs/src/get_started.md [hunk 14]
```diff
@@ -0,0 +1,104 @@
...
+
+---
+
+### Quick path
+
+```@raw html
+<ol>
+  <li>
+    Install Obliqua: set up the code and required dependencies.<br>
+    ➔ <a href="[https://proteus-framework.org/Obliqua/dev/how-to-guides/install.html](https://proteus-framework.org/Obliqua/dev/how-to-guides/install.html)">Installation guide</a>
```
Comment (nichollsh) [inline docs/src/get_started.md]: This link (and all of them below) are broken because you've mixed up markdown and HTML syntax.

File: docs/src/how-to-guides/config_file.md [hunk 15]
```diff
@@ -0,0 +1,241 @@
+
+# Configuration file
+
+Obliqua uses [TOML](https://toml.io/en/) to structure its configuration files. This page lists all parameters with their types and descriptions. For topic-specific parameter guides, see the configuration reference pages:
+
+
+
+For worked examples, see the [tutorials](https://proteus-framework.org/Obliqua/dev/tutorials/0d_test.html).
```
Comment (nichollsh) [inline docs/src/how-to-guides/config_file.md]: Another broken link

File: .gitignore [hunk 16]
```diff
@@ -11,4 +11,7 @@ __pycache__
 .vscode/
 .venv/
 .CondaPkg/
-Obliqua/
\ No newline at end of file
+Obliqua/
+coverage.*
+src/*.cov
```
Comment (nichollsh) [inline .gitignore]: You need to add `test/*.cov` here too (or just `*.cov`).

File: docs/src/tutorials/0d_test.md [hunk 17]
```diff
@@ -0,0 +1,195 @@
...
+        147739735943.7933,
+        1000000000.0
+    ],
+    "phi": [
+        0.0,
+        1.0
+    ],
+    "ncalc": 1000
+}
+```
```
Comment (nichollsh) [inline docs/src/tutorials/0d_test.md]: Please add comments to these lines which explain what each quantity is. The tutorial is meant to teach the user, but at the moment it doesn't really tell me what I am doing here...

File: docs/src/tutorials/0d_test.md [hunk 18]
```diff
@@ -0,0 +1,195 @@
...
+[ Info: Smoothing between indices 1 and 2 (r = 480000.0 to 995613.0 m)
+[ Info: Writing results to /out/0_obliqua.nc
+[ Info: Expected bulk heating: 2.9281844574776967e22
+[ Info: Obtained bulk heating: 0.0
+```
+
+First, Obliqua will load the interior data and the configuration file. Next it will decide on which modes to include in the calculation. In this case, we are using the full spectrum, which restricts to the same mode for all forcing frequencies. Next, the complex modulus profiles are smoothed to avoid sharp jumps in viscosity, which can cause numerical instabilities. Finally, the results are written to a NetCDF file in the `out` directory. The expected bulk heating is also printed to the terminal, which is the total tidal dissipation rate in Watts. However, this is not relevant for the full spectrum. The obtained bulk heating is also printed, which is the total radially integrated tidal dissipation rate from the heating profile in Watts, and should be zero for all zero-dimensional models.
+
+Let us now run the `examples/visualize_output.ipynb` notebook to visualize the results. The output should look like this:
```
Comment (nichollsh) [inline docs/src/tutorials/0d_test.md]: This notebook doesn't work for me because of the following error:
```
ModuleNotFoundError                       Traceback (most recent call last)
Cell In[1], [line 8](vscode-notebook-cell:?execution_count=1&line=8)
      4 import matplotlib.ticker as ticker
      5 from netCDF4 import Dataset
      6 
      7 from cmcrameri import cm
----> [8](vscode-notebook-cell:?execution_count=1&line=8) import scienceplots
      9 plt.style.use(['science', 'no-latex'])  # Use 'science' style without LaTeX rendering
     10 plt.rcParams.update({"font.size":15})

ModuleNotFoundError: No module named 'scienceplots'
```

In general, it would be great if the tutorials didn't rely on a notebook and (especially) if they didn't rely on Python. If this is really necessary, steps explaining how to install conda need to be included in the docs too...

**Review by nichollsh — CHANGES_REQUESTED** [review]
Thanks for this Marijn. I love the animation on the homepage! Great work. However, I have a number of critical suggestions which need to be addressed before this is ready. See my in-line comments below and higher-level comments in this review.

This is my "human" review, but I'll get Claude to take a pass, too.

### Tests

The tests pass on my laptop (Fedora 44 on x86, Julia 1.12.6). The coverage is 81.7% - which is very nice.

### Tutorial 0D

I managed to get the tutorial working, but it doesn't work as-described in the docs. The final plot looks as below, which does indeed reproduce the one you've provided.

Image

### Tutorial 1D

Doesn't work because the file `channel_showcase_updated.json` is missing from the repo. I tried using the `test_*.json` files instead, but they fail with a KeyError on "phi".

### Write NetCDF error

When writing the output netcdf files, I get the following error if the file already exists. Consider checking + warning + removing the file first, if it exists.

``log
[ Info: Writing results to /home/harrison/Projects/Obliqua/out/0_obliqua.nc
ERROR: NetCDF error: Permission denied (NetCDF error code: 13)
```

### Docs links

Many of the documentation links are broken. This is for various reasons including: incorrect HTML/Markdown syntax, linking directly to proteus-framework.org, typos, and some links are simply missing.

For example, the page `running_output.md` doesn't appear in the sidebar.

**Comment by nichollsh** [comment]
**Selected comments from Claude code review**, two things:

1. Undiscussed sign change in an untested fluid-body matrix — `src/common.jl:1044`

The 4x4 fluid-body `get_A!(A, ω, r, ρ, g, K, n; ...)` (docstring cites Korenaga (2025) Eq. 12) changed:
```
A[Y[2],Y[3]] = ρ*(n+1)*r_inv + n*(n+1)*ρ*g / (r^2 * ω^2)
```
to
```
A[Y[2],Y[3]] = ρ*(n+1)*r_inv - n*(n+1)*ρ*g / (r^2 * ω^2)
```
Unlike the other sign changes in this PR, this one isn't mentioned anywhere. It also has no callers anywhere in `src/` and is never exercised by any test — confirmed via the `.cov` coverage annotations, where this function's lines show no hit count at all while neighboring functions show large hit counts. A sign error here would never be caught by CI. Either add a test that pins it against Korenaga Eq. 12 (or against the same `Ic`-vs-`A` finite-difference consistency check used elsewhere in `test_common.jl`, if a matching `Ic` exists), or remove it if it's vestigial.

2. `solid_shell` patch leaks its placeholder moduli into the heating-map output — `src/solid1d_relax.jl:61-65`, `src/solid1d_mush_relax.jl:72-78`

When `patch=true` (set in `Obliqua.jl` whenever `solid_shell` is on and the segment is solid-relax/mush-relax with all layer viscosities ≤ 1e17 Pa·s), `solve_radial_system` does:
```
μ[1] = precc(1.47e11)
K[1] = precc(6.58e10)
```
directly on its `μ`/`K` arguments — an in-place mutation, not a copy. In `Obliqua.jl`, `run_solid1d_relax` and `run_solid1d_mush_relax` pass the very same `μc`/`κc` arrays into `get_heating_profile`/`get_heating_map` right after `compute_y` returns. Since the patch is never undone, the heating diagnostic at the patched (core-adjacent) layer is silently computed with these placeholder moduli instead of the real layer material, with nothing documenting that this happens. (It doesn't leak across time-steps — `resample_profiles`/`copy()` isolate the array per call — but it does corrupt that call's own heating output.)

Suggest operating on a local copy of `μ`/`K` (and `Kd`/`k`/`ϕ` in the mush case) inside the `patch` branch, or computing the heating profile/map from the un-patched moduli explicitly, or — if the patched-layer heating is actually meant to reflect the placeholder shell — documenting that explicitly.

File: docs/src/index.md (hunk 12 above)
Comment (MarijnJ0) [reply docs/src/index.md]: I wrapped it in a `<details>` block, so its hidden by default.

File: docs/src/tutorials/0d_test.md (hunk 18 above)
Comment (MarijnJ0) [reply docs/src/tutorials/0d_test.md]: I removed the scienceplots dep, it is now using proteus-mpl. Personally, I prefer to use mpl over plots.jl for all plotting perposes. Given this, I have added the relevant conda setup steps to the install page in the docs.

**Comment by MarijnJ0** [comment]
Thanks Harrison! 

I have added the requested fixes to the docs. Regarding the code, the unused 4x4 A matrix was added mainly for testing purposes, but since it doesn't add much to the relaxation based solvers, I decided to remove it completely. Potentially, it can be reintroduced at some point in the future for completeness. 

The issue with the patch leaking into the heating profile is now also fixed. Before reassigning the first entry in the array, the code now first creates a local copy.

I tried both tutorials and they seem to work, I added the new data file so this should also be fixed now. 

For plotting I kept the python notebook, as I think it is more convenient to work with than plots.jl. For this, there is now also a conda install section in the docs.

If you have any further comments do let me know!

**Review by nichollsh — APPROVED** [review]
Great - thank you, @MarijnJ0, for implementing my suggestions and fixes. 

This seems to work as described on my laptop. The tests still pass with good code coverage.

The 1D tutorial now works, and I get the plot below, which matches the documentation.

Image

I will mention that the example notebook contains some "empty" plots of heating vs lon/lat. This isn't mentioned in the tutorial at the moment, so I suppose they are remnants or intended for future expansion?

Happy for this to be merged. Well done!

**Comment by nichollsh** [comment]
This PR is currently blocked by @timlichtenberg's previous review requesting changes. I have given it a look, but @timlichtenberg needs to re-review or dismiss the old one before it can be merged.

**Comment by MarijnJ0** [comment]
Thanks again, great to hear it is working! 

FYI: The empty plots for the heatmaps only get populated when `store_3D    = true`, otherwise the angular grid gets averaged over the solid angle, and we only end up with the heating profile.

**Comment by MarijnJ0** [comment]
These two commits also add the relevant fixes for the `solid1d_mush_relax` model and the inertial core solution vector. 

I verified the mush model against the description in overleaf, and they match. The final part of the derivation is now in overleaf, and is more of a simplification step.

Additionally, to fix the inertial core solution vector, I found a typo in the original reference: https://www.aanda.org/articles/aa/full_html/2021/06/aa39433-20/aa39433-20.html see inline comments. This now adds the fully general solid solution, verified against A through finite-difference checks. This simplifies in the fluid limit to the `inertial-liquid` solution (https://www.sciencedirect.com/science/article/pii/S0019103525003045), which was originally called `inertial`. This is also verified against A through finite-difference checks.

The core solution vectors relate roughly as follows:

- `"inertial"` reduces to `"inertial-liquid"` as μ -> 0: an oscillating
  solid without shear rigidity is an oscillating fluid.
- `"inertial"` reduces to `"solid"` as ω -> 0 and K -> infinity: the
  static, incompressible limit of the general oscillating solid recovers
  the classical elastic solution.
- `"solid"` (μ -> 0) reduces to `"liquid"` (ω = 0): incompressible, static
  elasticity without shear rigidity is a hydrostatic fluid.
- `"inertial-liquid"` reduces to `"liquid"` as ω -> 0: a slowly
  oscillating fluid approaches hydrostatic equilibrium.

**Review by timlichtenberg — APPROVED** [review]
This is much improved. The new validation layer is exactly the kind of check I wanted to see: the core basis vectors are now checked against the motion matrix by finite-difference residual and against the Love 1911 closed form, and the 1D solid modules are checked against the 0D homogeneous-sphere solution in the analytical limit. That moves the physics from self-consistent to validated against independent references.

I ran the full suite on `md/phys_tests` (Julia 1.12): 591 tests pass, 0 failures.

Good work improving the PR. Approving.
